http.md

```diff
@@ -0,0 +1,342 @@
+
+## Definitions
+
+- Bundle: Bundles are a list of transactions that execute sequentially and atomically. “All or nothing” so to speak. This means that a user can send a bundle that contains multiple transactions and guarantee that they are all executed one after the other and the bundle succeeds only if all individual transactions succeed.
+- Hash: A SHA-256 hash of a chunk of data.
+- Pubkey: The public key of a Ed25519 key-pair.
+- Tip Account: List of accounts to which a tip can be paid for processing the bundles
+- Transaction: A list of Solana instructions signed by a client keypair to authorize those actions.
+- Signature: An Ed25519 signature of transaction's payload data including instructions. This can be used to identify transactions.
+
+## Configuring State Commitment
+
+For preflight checks and transaction processing, Solana nodes choose which bank
+state to query based on a commitment requirement set by the client. The
+commitment describes how finalized a block is at that point in time. When
+querying the ledger state, it's recommended to use lower levels of commitment
+to report progress and higher levels to ensure the state will not be rolled back.
+
+In descending order of commitment (most finalized to least finalized), clients
+may specify:
+
+- `"finalized"` - the node will query the most recent block confirmed by supermajority
+  of the cluster as having reached maximum lockout, meaning the cluster has
+  recognized this block as finalized
+- `"confirmed"` - the node will query the most recent block that has been voted on by supermajority of the cluster.
+  - It incorporates votes from gossip and replay.
+  - It does not count votes on descendants of a block, only direct votes on that block.
+  - This confirmation level also upholds "optimistic confirmation" guarantees in
+    release 1.3 and onwards.
+- `"processed"` - the node will query its most recent block. Note that the block
+  may still be skipped by the cluster.
+
+For processing many dependent transactions in series, it's recommended to use
+`"confirmed"` commitment, which balances speed with rollback safety.
+For total safety, it's recommended to use`"finalized"` commitment.
+
+### Example
+
+The commitment parameter should be included as the last element in the `params` array:
+
+```bash
+curl http://localhost:8899 -X POST -H "Content-Type: application/json" -d ' { "jsonrpc": "2.0", "id": 1, "method": "getBundleStatuses", "params": [ "bundle", { "commitment": "finalized" } ] }'
+```
+
+#### Default
+
+If commitment configuration is not provided, the node will default to `"finalized"` commitment
+
+Only methods that query bank state accept the commitment parameter. They are indicated in the API Reference below.
+
+#### RpcResponse Structure
+
+Many methods that take a commitment parameter return an RpcResponse JSON object comprised of two parts:
+
+- `context` : An RpcResponseContext JSON structure including a `slot` field at which the operation was evaluated.
```

Comment:
Link to RpcResponseContext structure or docs.rs?

Reply:
This is the same as in https://docs.solana.com/api/http#rpcresponse-structure I used the notes relevant to us.
It's mostly telling that in some cases, we provide the context that provide a slot field and whatever value that is returned. Sort of like

`### Response

```json
{
    "jsonrpc": "2.0",
    "result": {
        "context": {
            "slot": 82
        },
        "value": [
            {
                "bundle": {
                    "signatures": [
                        "3Eq21vXNB5s86c62bVuUfTeaMif1N2kUqRPBmGRJhyTA",
                        "2nBhEBYYvfaAe16UMNqRHre4YNSskvuYgx3M6E4JP1oDYvZEJHvoPzyUidNgNX5r9sTyN1J9UxtbCXy2rqYcuyuv"
                    ]
                },
                "slot": 1234,
                "confirmationStatus": "finalized",
                "err": "null"
            }
        ]
    },
    "id": 1
}`